File: crates/psychevo-gateway/tests/fixtures/fake_acp_wrong_version.py
#!/usr/bin/env python3
import json
import sys

log_path = sys.argv[1]


def send(value):
    print(json.dumps(value), flush=True)


for line in sys.stdin:
    if not line.strip():
        continue
    message = json.loads(line)
    with open(log_path, "a", encoding="utf-8") as log_file:
        log_file.write(json.dumps({"method": message.get("method")}) + "\n")
    if message.get("method") == "initialize":
        send(
            {
                "jsonrpc": "2.0",
                "id": message.get("id"),
                "result": {"protocolVersion": 2, "agentCapabilities": {}},
            }
        )

File: crates/psychevo-gateway/tests/fixtures/fake_acp_wrong_version.js
#!/usr/bin/env node
const fs = require("node:fs");
const readline = require("node:readline");
const logPath = process.argv[2];

readline.createInterface({ input: process.stdin, crlfDelay: Infinity }).on("line", (line) => {
  if (!line.trim()) return;
  const message = JSON.parse(line);
  fs.appendFileSync(logPath, `${JSON.stringify({ method: message.method })}\n`, "utf8");
  if (message.method === "initialize") {
    process.stdout.write(`${JSON.stringify({ jsonrpc: "2.0", id: message.id, result: { protocolVersion: 2, agentCapabilities: {} } })}\n`);
  }
});
